tests/test_salesforce_api_total_quota.py

```diff
@@ -0,0 +1,57 @@
+from sfbase import SFBaseTest
+from tap_tester import runner, menagerie, LOGGER, connections
+
+
+class SFAPIQuota( SFBaseTest):
+    """
+    https://jira.talendforge.org/browse/TDL-23431
+    This testcase makes sure we are able to configure the tap with specific
+    total quota for apis and verifies if tap is working as per the
+    api quota limit set.
+    """
+
+    """
+    Set a start date well past so that we have enough data to sync to hit the limit
+    set a low quota to make the sync fail
+    some streams are excluded as they don't have any data
+    """
+
+    start_date = '2000-11-23T00:00:00Z'
+    total_quota = '1'
+    streams_to_exclude = {
+             'DatacloudAddress',
+             'DatacloudCompany',
+             'DatacloudContact',
+             'DatacloudDandBCompany',
+             'DatacloudOwnedEntity',
+             'DatacloudPurchaseUsage',
+             'FieldSecurityClassification',
+             'ServiceAppointmentStatus',
+             'WorkOrderLineItemStatus',
+             'WorkOrderStatus',
+             'ShiftStatus',
+             'WorkStepStatus',
+        }
+
+    @staticmethod
+    def name():
+        return "tt_sf_api_quota_total"
+
+    def streams_to_test(self):
+        return self.expected_streams().difference( self.streams_to_exclude )
```

Comment:
```suggestion
        return self.expected_streams().difference(self.streams_to_exclude)
```

Reply:
sorry, missed some comments. I removed some more white spaces and hopefully I took care of all of them